doc/sphinx/practical-tools/coqide.rst

```diff
@@ -327,3 +327,123 @@ If you choose something else than UTF-8, then missing characters will
 be written encoded by `\x{....}` or `\x{........}` where each dot is
 an hexadecimal digit: the number between braces is the hexadecimal
 Unicode index for the missing character.
+
+.. _coqide-debugger:
+
+Debugger
+--------
+
+Version 8.14 introduces a preliminary visual debugger for Ltac tactics within
+CoqIDE.  It supports setting breakpoints visually, automatically
+displaying the stopping point in the source code with "continue",
+"step over" "step in" and "step out" operations.  It does not yet
+display the call stack or variable values.
+
+The debugger is based on the :ref:`Ltac debugger <interactive-debugger>`.
+We'd like to eventually support other scripting facilities such as Ltac2,
+but that will be addressed later.  (Since Ltac2 shares tactic evaluation
+logic with Ltac, it may already work to some extent.)
+
+Even though the debugger is incomplete, we believe it can already provide
+some value to users.  You may encounter bugs or usability issues.  The behavior
+and user interface will evolve as the debugger is completed.  Please feel free
+to suggest changes and improvements.
+
+GUI
+~~~
+
+This screenshot shows the debugger stopped at a breakpoint in the Ltac tactic
+`my_tac`.  Breakpoints are shown with a red background and the stopping point is
+shown with a dark blue background.  `Set Ltac Debugger.` enables stopping in the
+debugger.
+
+  .. image:: ../_static/debugger.png
+     :alt: CoqIDE Debugger
+
+You can control the debugger entirely with function keys.  Some messages are
+shown in the Messages panel.  You can type commands in that panel when it shows
+the debug prompt, but it's limited to the
+:ref:`debugger commands <interactive-debugger>` available in coqtop.
+
+The script is not editable while Coq is processing tactics or stopped
+in the debugger.  The "in progress" slider at the bottom edge of the window will
+indicate Coq is busy when the debugger is stopped (e.g. at a breakpoint).
+
+The function keys are listed, for the moment, with one exception, at the bottom
+of the `View` menu:
+
+  .. image:: ../_static/debugger-menu.png
+     :alt: Debugger function keys
+
+Toggle breakpoint (F8)
+  Position the cursor on the first character of the tactic name, then press F8.
+  Press again to remove the breakpoint.  Breakpoints at any point you can single-step
+  to are valid; breakpoints at other locations are not.  (The debugger engine may
+  support breakpoint in some other places that are not the beginning of tactics.)
+
```

Comment:
Does "single step" here mean within the debugger, or in normal tactic evaluation?  (i.e., is it about tactics you type, or about periods in tactic scripts?)  Aren't most places considered "not the beginning of tactics"?  Or do you have something particular in mind here?

Reply:
"single-step" meaning you can reach it with "Step In" (F10).  But now that I think about it some more, I think the last two sentences are confused and I'll drop them.

You can set a breakpoint on any syntactically elementary tactic, e.g. on `bar` in `foo; bar.`